### PR TITLE
Feature/284781  lookup country

### PR DESCRIPTION
**Pull request by whitewaterdesign**

### CODE REVIEW

File: app/lib/format-helpers.js [hunk 1]
```diff
@@ -14,6 +14,23 @@ const formatAddress = (address) => {
...
+const countryCodeMap = {
+  E: 'England',
+  S: 'Scotland',
+  W: 'Wales'
+}
+
+const mapOsCountryCodeToCountry = (osPlacesCountryCode) => {
+  const country = countryCodeMap[osPlacesCountryCode]
+
+  if (country !== undefined) {
```
Comment (jbarnsley10) [inline app/lib/format-helpers.js:26]: Does this handle null too?

Comment (jbarnsley10) [reply app/lib/format-helpers.js:26]: A neat way is to do:
if (country) {
- this handles undefined and null in one

File: app/models/cdo/create/address.js [hunk 2]
```diff
@@ -46,34 +53,32 @@ function ViewModel (address, errors) {
...
       const name = error.path[0]
 
-      const prop = this.model[name]
-
-      if (prop !== undefined) {
-        prop.errorMessage = {
-          text: error.message
-        }
-
+      if (name === 'country') {
```
Comment (jbarnsley10) [inline app/models/cdo/create/address.js:64]: Not sure why you have to deal with 'county' explicitly. Does is not follow the generic error flow?

File: app/models/cdo/create/address.js [hunk 3]
```diff
@@ -46,34 +53,32 @@ function ViewModel (address, errors) {
...
-      autocomplete: forms.preventAutocomplete,
-      attributes: { maxlength: '30' }
-    },
+    countries: mapToCountrySelector(countries, validationError, address.country),
     errors: []
   }
 
-  if (errors) {
-    for (const error of errors.details) {
+  if (validationError) {
```
Comment (jbarnsley10) [inline app/models/cdo/create/address.js:60]: If you're changing the name of 'errors' to '#validationError', I don;t have a problem - but it should be plural (i.e. 'validationErrors') as there may be more than one in the array

File: app/models/cdo/create/country.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+const { routes } = require('../../../constants/cdo/owner')
```
Comment (jbarnsley10) [inline app/models/cdo/create/country.js]: Don;t think this file is needed anymore

File: app/models/mappers/countries.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+   */
+  const countriesConcat = countries.map(country => ({
+    value: country,
+    text: country
+  }))
+
+  /**
+   * @type {CountrySelectorItem[]}
+   */
+  const items = [{ text: 'Choose country', value: '', selected: true }].concat(countriesConcat)
```
Comment (jbarnsley10) [inline app/models/mappers/countries.js:67]: Think it should be 'Choose a country' (with 'a' in the middle)

File: app/schema/portal/cdo/country.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+const Joi = require('joi')
```
Comment (jbarnsley10) [inline app/schema/portal/cdo/country.js:1]: Don;t think this file is needed anymore

File: app/views/cdo/create/address.njk [hunk 7]
```diff
@@ -48,6 +49,8 @@
         
           {{ govukInput(model.postcode) }}
 
+          {{ govukSelect(model.countries) }}
```
Comment (jbarnsley10) [inline app/views/cdo/create/address.njk:52]: Being picky this name should be 'model.country' as it's a single field (select field)

File: app/views/cdo/create/confirm-all-details.njk [hunk 8]
```diff
@@ -58,7 +58,19 @@
...
             </div>
-
+            <div class="govuk-summary-list__row">
+              <dt class="govuk-summary-list__key">
+                Country
+              </dt>
+              <dd class="govuk-summary-list__value">
+                {{ model.summary.owner.country }}
+              </dd>
+              <dd class="govuk-summary-list__actions">
```
Comment (jbarnsley10) [inline app/views/cdo/create/confirm-all-details.njk:68]: Can we remove the actions column, and set the row attribute to --no_actions

**Review by jbarnsley10 — CHANGES_REQUESTED** [review]
Good stuff. Just a few minor comments

File: app/models/cdo/create/country.js (hunk 4 above)
Comment (whitewaterdesign) [reply app/models/cdo/create/country.js]: Yeah, this file was deleted on a subsequent commit

File: app/schema/portal/cdo/country.js (hunk 6 above)
Comment (whitewaterdesign) [reply app/schema/portal/cdo/country.js:1]: thought this was deleted

File: app/models/cdo/create/address.js (hunk 3 above)
Comment (whitewaterdesign) [reply app/models/cdo/create/address.js:60]: As discussed, this is a Joi.ValidationError object containing an array of details.

File: app/models/cdo/create/address.js (hunk 2 above)
Comment (whitewaterdesign) [reply app/models/cdo/create/address.js]: the model.country change resolves this perfectly.  Thanks again.

**Review by jbarnsley10 — APPROVED** [review]
Nice job

**Pull request merged**